
this was triggered due to this upstream change:
https://git.samba.org/?p=samba.git;a=commit;h=83fbdc81cdfe6c018bb97d2d482ca09389b2c7af

heimdal 1.5 changed the api:
https://github.com/heimdal/heimdal/commit/c757eb7fb04a9b0ca883ddb72c1bc75bf5d814f3

proper fix means the configure script should check which to use

--- ./source4/kdc/kdc.c.orig
+++ ./source4/kdc/kdc.c
@@ -967,9 +967,9 @@
 	 * The old behavior in the _kdc_get_preferred_key()
 	 * function is use_strongest_server_key=TRUE.
 	 */
-	kdc->config->as_use_strongest_session_key = false;
+	kdc->config->tgt_use_strongest_session_key = false;
 	kdc->config->preauth_use_strongest_session_key = false;
-	kdc->config->tgs_use_strongest_session_key = false;
+	kdc->config->svc_use_strongest_session_key = false;
 	kdc->config->use_strongest_server_key = true;
 
 	/* Register hdb-samba4 hooks for use as a keytab */
